Provide the basic api and encryption

beercanx opened the pull request:
This branch will be updated with the changes required to implement a basic api and cipher model.

The API will have a [http://swagger.io/specification](http://swagger.io/specification) provided for it, which will probably manually created as it makes it easier to work out what to create and test.

The API will be built with the idea that it will be used in a client-server and not a server-server setup, shall support at least json so its easier to possibly build the GUI in Angular or some other javascript framework.

**Tasks**
- [x] Basic create operations
- [ ] Generate keys and secure passwords, sharing password keys with bags
- [ ] Generate keys and secure bags, sharing bag keys with users
- [ ] Update create user to require password and generate keys to secure the user
- [ ] Create user authentication end point to provide client side session
- [ ] Update create bag and password to require authentication session

**NOTE:** This branch will not be ready for merging any time soon.

[![Coverage Status](https://coveralls.io/repos/github/beercan1989/secure-my-bacon/badge.svg?branch=basic-api-with-encryption)](https://coveralls.io/github/beercan1989/secure-my-bacon?branch=basic-api-with-encryption)

[![Dependency Status](https://www.versioneye.com/user/projects/57fd171f9907da00390c41a2/badge.svg?style=flat-square)](https://www.versioneye.com/user/projects/57fd171f9907da00390c41a2)

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/beercan1989/secure-my-bacon/1)

<!-- Reviewable:end -->

beercan1989 commented on base/src/test/java/uk/co/baconi/secure/base/pagination/PaginatedResultTest.java:
```diff
@@ -0,0 +1,70 @@
+package uk.co.baconi.secure.base.pagination;
+
+import org.junit.Before;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.junit.Before'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2175852265/issues/source?bid=2998592&fileBranchId=3163365#l3)

beercan1989 commented on base/src/test/java/uk/co/baconi/secure/base/pagination/PaginatedResultTest.java:
```diff
@@ -0,0 +1,70 @@
+import uk.co.baconi.secure.base.BaseUnitTest;
+import uk.co.baconi.secure.base.bag.Bag;
+import uk.co.baconi.secure.base.lock.AsymmetricLock;
+import uk.co.baconi.secure.base.user.User;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'uk.co.baconi.secure.base.user.User'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2175852265/issues/source?bid=2998592&fileBranchId=3163365#l12)

beercan1989 commented on base/src/test/java/uk/co/baconi/secure/base/pagination/PaginatedResultTest.java:
```diff
@@ -0,0 +1,70 @@
+import org.springframework.data.domain.Page;
+import uk.co.baconi.secure.base.BaseUnitTest;
+import uk.co.baconi.secure.base.bag.Bag;
+import uk.co.baconi.secure.base.lock.AsymmetricLock;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'uk.co.baconi.secure.base.lock.AsymmetricLock'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2175852265/issues/source?bid=2998592&fileBranchId=3163365#l11)

beercan1989 commented on base/src/test/java/uk/co/baconi/secure/base/pagination/PaginatedResultTest.java:
```diff
@@ -0,0 +1,70 @@
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.data.domain.Page;
+import uk.co.baconi.secure.base.BaseUnitTest;
+import uk.co.baconi.secure.base.bag.Bag;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'uk.co.baconi.secure.base.bag.Bag'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2175852265/issues/source?bid=2998592&fileBranchId=3163365#l10)

beercan1989 commented on base/src/main/java/uk/co/baconi/secure/base/BaseConfiguration.java:
```diff
@@ -27,9 +28,8 @@
-import org.springframework.data.neo4j.server.Neo4jServer;
-import org.springframework.data.neo4j.server.RemoteServer;
 import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.util.StringUtils;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.springframework.util.StringUtils'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2652286677/issues/source?bid=2998592&fileBranchId=3163365#l32)

beercan1989 commented on base/src/test/java/uk/co/baconi/secure/base/BaseConfigurationTest.java:
```diff
@@ -0,0 +1,96 @@
+
+import org.apache.commons.codec.binary.Base64;
+import org.junit.Test;
+import org.neo4j.ogm.config.Configuration;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.neo4j.ogm.config.Configuration'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2653168778/issues/source?bid=2998592&fileBranchId=3163365#l21)

beercan1989 replied:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.neo4j.ogm.config.Configuration'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2653200075/issues/source?bid=2998592&fileBranchId=3163365#l21)

beercan1989 replied:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.neo4j.ogm.config.Configuration'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2653248427/issues/source?bid=2998592&fileBranchId=3163365#l21)

beercan1989 commented on api/src/main/java/uk/co/baconi/secure/api/user/UserEndpoint.java:
```diff
@@ -16,33 +16,89 @@
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import uk.co.baconi.secure.base.bag.Bag;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'uk.co.baconi.secure.base.bag.Bag'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2653716498/issues/source?bid=2998592&fileBranchId=3163365#l27)

beercan1989 commented on api/src/main/java/uk/co/baconi/secure/api/lock/SymmetricLockEndpoint.java:
```diff
@@ -16,34 +16,58 @@
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import uk.co.baconi.secure.api.bag.BagEndpoint;
+import uk.co.baconi.secure.base.bag.Bag;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'uk.co.baconi.secure.base.bag.Bag'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2741326092/issues/source?bid=2998592&fileBranchId=3163365#l31)

beercan1989 commented on api/src/main/java/uk/co/baconi/secure/api/lock/SymmetricLockEndpoint.java:
```diff
@@ -16,34 +16,58 @@
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import uk.co.baconi.secure.api.bag.BagEndpoint;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'uk.co.baconi.secure.api.bag.BagEndpoint'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2741326092/issues/source?bid=2998592&fileBranchId=3163365#l30)

beercan1989 commented on api/src/main/java/uk/co/baconi/secure/api/lock/AsymmetricLockEndpoint.java:
```diff
@@ -16,33 +16,57 @@
 import org.springframework.web.bind.annotation.RestController;
 import uk.co.baconi.secure.base.lock.AsymmetricLock;
 import uk.co.baconi.secure.base.lock.AsymmetricLockGraphRepository;
+import uk.co.baconi.secure.base.lock.SymmetricLock;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'uk.co.baconi.secure.base.lock.SymmetricLock'](https://www.codacy.com/app/beercan1989/secure-my-bacon/file/2741326091/issues/source?bid=2998592&fileBranchId=3163365#l32)

beercanx committed:
Using more annotation driven application configuration.

beercanx committed:
Changes to stop using deprecated code and use the new improved version.

beercanx committed:
Changes to stop using deprecated code and use the new improved version.

beercanx committed:
Revert "Removal of unit test code that's no good."

This reverts commit 2d7b028db2ace0a9833d4bd9cd665233222c4507.

beercanx committed:
Revert "Removal of unit test code that's no good."

This reverts commit 2d7b028db2ace0a9833d4bd9cd665233222c4507.

beercanx committed:
Improving code coverage by removing uneseccery methods, also improving trace messages.

beercanx committed:
Improving code coverage by removing uneseccery methods, also improving trace messages.

beercanx committed:
More lombok conversions.

beercanx committed:
More lombok conversions.

beercanx committed:
Updating find all method named to some thing sensible.

beercanx committed:
Updating find all method named to some thing sensible.

beercanx committed:
Enabling random free port for integration tests.

beercanx committed:
Enabling random free port for integration tests.

beercanx committed:
Updated validation to include checks for blank user names.

beercanx committed:
Updated validation to include checks for blank user names.

beercanx committed:
Updated BagEndpoint to start enabling Bag creation
... along with everything else involed in that actions.

beercanx committed:
Updated BagEndpoint to start enabling Bag creation
... along with everything else involed in that actions.

beercanx committed:
Making varaible final and improving logging.

beercanx committed:
Making varaible final and improving logging.

beercanx committed:
Configuration to create required model constrains on start up.

beercanx committed:
Configuration to create required model constrains on start up.

beercanx committed:
Removed enterprise only constraint.

beercanx committed:
Removed enterprise only constraint.

beercanx committed:
Removed test to try and cover main method in ApiApplication, as now needs a Neo4J server.

beercanx committed:
Removed test to try and cover main method in ApiApplication, as now needs a Neo4J server.

beercanx committed:
Changes to tests after unique constraint on User and Group names.

beercanx committed:
Changes to tests after unique constraint on User and Group names.

beercanx committed:
Silly mistake in naming Bag a Group.

beercanx committed:
Silly mistake in naming Bag a Group.

beercanx committed:
Removing unused import

beercanx committed:
Removing unused import

beercanx committed:
Fixes to tests after correctly setting unique constraint on Bags

beercanx committed:
Fixes to tests after correctly setting unique constraint on Bags

beercanx committed:
Added docker specific properties

beercanx committed:
Added docker specific properties

beercanx committed:
Script to help reset the default neo4j password.

beercanx committed:
Script to help reset the default neo4j password.

beercanx committed:
Property changes to enable features.

Also added a property to handle changing the default password

beercanx committed:
Property changes to enable features.

Also added a property to handle changing the default password

beercanx committed:
Changes to check for enabled flags and to reset neo4j password before running other schema changes.

beercanx committed:
Changes to check for enabled flags and to reset neo4j password before running other schema changes.

beercanx committed:
Changes to help garuentee

beercanx committed:
Changes to help garuentee

beercanx committed:
Removing feature to auto change default admin pwd

Just too much pain involved.

beercanx committed:
Removing feature to auto change default admin pwd

Just too much pain involved.

beercanx committed:
Changed docker to require maven run locally already

beercanx committed:
Changed docker to require maven run locally already

beercanx committed:
Removing unused code

beercanx committed:
Removing unused code

beercanx committed:
Added the ability to find bags by name

beercanx committed:
Added the ability to find bags by name

beercanx committed:
Added new endpoint to create a password

beercanx committed:
Added new endpoint to create a password

beercanx committed:
Removed unused import.

beercanx committed:
Removed unused import.

beercanx committed:
Update to test logging config.

beercanx committed:
Update to test logging config.

beercanx committed:
Reorganising test code

beercanx committed:
Reorganising test code

beercanx committed:
Added tests to cover basic creation of bags and passwords

beercanx committed:
Added tests to cover basic creation of bags and passwords

Commit:
Quick Codacy fixes, keeping code clean-ish.

Commit:
Quick Codacy fixes, keeping code clean-ish.

Commit:
Added test controller to enable testing exception handling tests.

Commit:
Added test controller to enable testing exception handling tests.

beercanx committed:
Added versioneye as a test

beercanx committed:
Added versioneye as a test

beercanx committed:
Updated with versioneye id and travis to update

beercanx committed:
Updated with versioneye id and travis to update

beercanx committed:
Backed out complex versioneye integration

beercanx committed:
Backed out complex versioneye integration

beercanx committed:
Correcting ReadMe layout

beercanx committed:
Correcting ReadMe layout

beercanx committed:
Fix to versioneye project id in badge

beercanx committed:
Fix to versioneye project id in badge

beercanx committed:
Added ability to run with an embedded neo4j server

beercanx committed:
Added ability to run with an embedded neo4j server

beercanx committed:
Moved common tests into their own package.

Also includes a some new find by id and name common tests.

beercanx committed:
Moved common tests into their own package.

Also includes a some new find by id and name common tests.

beercanx committed:
Updated tests to better test get by id and name.

beercanx committed:
Updated tests to better test get by id and name.

beercanx committed:
New exception handling and not found exception

beercanx committed:
New exception handling and not found exception

beercanx committed:
Check for null user on find by id or name.

beercanx committed:
Check for null user on find by id or name.

beercanx committed:
Implement find by id and name with not found checks.

beercanx committed:
Implement find by id and name with not found checks.

beercanx committed:
Fixes to logging of exception responses

as well as code coverage!

beercanx committed:
Fixes to logging of exception responses

as well as code coverage!

beercanx committed:
Quick lint fix

beercanx committed:
Quick lint fix

beercanx committed:
Updating test to not just throw a vanilla exception

so that we can shut up the linting

beercanx committed:
Updating test to not just throw a vanilla exception

so that we can shut up the linting

beercanx committed:
Update README.md

beercanx committed:
Update README.md

beercanx committed:
Inintial taughts on how to record the cipher type.

Another alternative is to store it all in the cipher text block, but
that can make it harder to query by.

beercanx committed:
Inintial taughts on how to record the cipher type.

Another alternative is to store it all in the cipher text block, but
that can make it harder to query by.

beercanx committed:
Ideas added over password based ciphers

beercanx committed:
Ideas added over password based ciphers

Commit:
Playing with adding encryption onto password creation.

Commit:
Playing with adding encryption onto password creation.

beercanx committed:
Fixes to tests after changes to places that are encrypted

beercanx committed:
Merge branch 'basic-api-with-encryption' of github.com:beercan1989/secure-my-bacon into basic-api-with-encryption

# Conflicts:
#	api/src/integration/java/uk/co/baconi/secure/api/integrations/IntegrationApiApplication.java
#	base/src/integration/java/uk/co/baconi/secure/base/lock/SymmetricGraphRepositoryIT.java
#	base/src/integration/java/uk/co/baconi/secure/base/password/PasswordGraphRepositoryIT.java
#	base/src/main/java/uk/co/baconi/secure/base/lock/AsymmetricLock.java
#	base/src/test/java/uk/co/baconi/secure/base/lock/SymmetricLockTest.java
#	base/src/test/java/uk/co/baconi/secure/base/password/PasswordTest.java

beercanx committed:
New service to handle converting Strings to bytes

beercanx committed:
New cipher exceptions that are expected happen

at least ocassionally so they are checked exceptions

beercanx committed:
Updated symmetric cipher with better padding

as well as key generator type

beercanx committed:
New unchecked exception for errors from cipher config

beercanx committed:
Register BouncyCastle on start up

beercanx committed:
Symmetric generator for keys and params

beercanx committed:
Symmetric cipher service

beercanx committed:
Updates to password endpoint to use changes in ciphers

beercanx committed:
Who doesn't love comments?

beercanx committed:
Updates after cipher type enum refactor

beercanx committed:
Fixes to tests after using encryption.

beercanx committed:
Added tests for new exceptions added for cipher operations.

beercanx committed:
Added failing test for CharsetCodec.

Appears that strings over 9 characters cause a null byte to be added.

beercanx committed:
Lint fixes: moved parameter static method to bottom of test.

beercanx committed:
Added fix for a problem where encoding certain strings it gains a null char.

beercanx committed:
Added more test data above the ASCII range.

beercanx committed:
Auto lint fixes.

Commit:
Added GCM type.

Commit:
Added example gcm decryption.

Commit:
Example into test. Formatting changes.

beercanx committed:
Updating spring boot.

beercanx committed:
Updating spring fox version.

Commit:
New graph repository call for password finding.

Commit:
Fix to nullpointer problem in tests.

beercanx committed:
Adding ability to find passwords for a user.

beercanx committed:
Fixed query to find password by user using id's.
Deprecated ID useage until the question of whether neo4js default ID's should be used in queries.

beercanx committed:
Added new getPasswordsByUser endpoint.
Also moved to using a users name instead of ID.

beercanx committed:
Extending base models with UUID for passwords. Deprecated the internal id from being used via graph repositories.

beercanx committed:
Added new endpoints and updated endpoints with changes to Password and use of UUID as the unique field on a Password node.

beercanx committed:
Removed neo4j id's from json models.

beercanx committed:
Removal of get by id endpoints. Move to use similar bnamed endpoints on Password get by's.

beercanx committed:
Updating NewRelic version.

beercanx committed:
Added PowerMock to enable testing of the mian method in ApiApplication.

beercanx committed:
Fix for handling non existent passwords by UUID.
Also includes new tests for password by UUID.
Along with changes to some test layouts.

beercanx committed:
Change to use pagination on getting lists of passwords for a user.

beercanx committed:
Updating spring boot version

beercanx committed:
Changes to the password repository as pageable isn't supported yet for custom queries.

beercanx committed:
IntegrationTest tidy up. Some tests for get passwords for user.

beercanx committed:
More tests for the passwords for user endpoint.

beercanx committed:
Fix to a silly mistake ater a copy n paste of a test.

beercanx committed:
Creating ability to use specific filtered resoures

beercanx committed:
Updating neo4j version.

Also includes fix for docker for setting the default user's password.

beercanx committed:
Ability to retry the schema creation on start up.

Mostly added to aid in the start up process when used in docker-compose.

beercanx committed:
Changing vagrant config to support Hyper-V

for windows where hyper-v has been enabled and VirtualBox cannot be used
as a result.

beercanx committed:
[Vagrant] - Disable IPv6 to improve provision times

and also disable sync folder mounting as it currently doesn't work nice
with Hyper-V on Windows 10.

beercanx committed:
[Vagrant] It appears neo4j 3.0.2 is not available in the deb repo.

beercanx committed:
[Vagrant] Making sure curl is installed.

beercanx committed:
[Vagrant] Updating method used to restart Neo4j

if the init.d script is there use it else we'll assume its systemd.

beercanx committed:
Docker compose to compile on request.

Also kept the manua compile and added a maven central docker build file.

beercanx committed:
Including new docker build files.

beercanx committed:
Fixing SSL issue on alpine docker and wget

beercanx committed:
Added JAVA_OPTS ability to docker

beercanx committed:
Centralising of dependency version management.

Also includes updates to the recent versions of the current
dependencies.

beercanx committed:
Updates for latest Neo4j OGM version...

including a refactor of and extra javadoc on the Neo4j properties.

beercanx committed:
Scripts to setup various neo4j servers in CI.

beercanx committed:
Fix to the on tag build its now jar not war.

Also includes changes to run the integration tests over a matrix of
Neo4j situations.

beercanx committed:
Log when the transaction manager is created.

beercanx committed:
Tear down of repositories before each test.

beercanx committed:
Ignoring neo4j folder and archive for CI.

beercanx committed:
Not explicetly setup test data for api before each test.

beercanx committed:
Fixing CI setup script to correctly skip on embedded

beercanx committed:
[CI] Explicitly setting the default neo4j password

beercanx committed:
[CI] Making neo4j setup script executable.

beercanx committed:
[CI] Switching to docker in Travis as manual mode didn't startup Neo4j after 13minutes.

beercanx committed:
[CI] Making Neo4j setup script for docker executable.

beercanx committed:
[Heroku] Fix to enable Heroku bundle builds to work.

beercanx committed:
[CI] Adding debug printing to neo4j setup scripts.

beercanx committed:
[CI] Correcting env conig in Travis to resolve correctly.

beercanx committed:
[CI] An attempt to simplify the Travis build matrix.

beercanx committed:
Revert "[CI] An attempt to simplify the Travis build matrix."

This reverts commit a01ee6fb1243b2e5755118304cd74a21f1a40e16.

beercanx committed:
[Heroku] Fixing broken properties after a refactor.

beercanx committed:
[Heroku] Adding a test to verify when the Heroku profile is enabled that it has been configured correctly.

beercanx committed:
Revert "[Heroku] Adding a test to verify when the Heroku profile is enabled that it has been configured correctly."

This reverts commit 1c85bc3b15eeaf71a2376363ecc877f5df045435.

beercanx committed:
[CI] Reducing the build matrix to speed up builds.

beercanx committed:
[Heroku] Triggering a deploy with the new stack.

Heroku-16 instead of Cedar-14.

beercanx committed:
Updating to the latest BouncyCastle version.

beercanx committed:
Resolving a spring fox deprecation.

beercanx committed:
Removal of unused imports.

beercanx committed:
[Dependency] Updating lombok to latest version.

beercanx committed:
[CI] Updating travis to test new version of neo4j and always the latest one.

beercanx committed:
[CI] Correcting docker version latest to be lowercase.

beercanx committed:
[Heorku] Configuration fix for running on Heroku as the graphenedb environment variables changed names.

beercanx committed:
Revert "[Heorku] Configuration fix for running on Heroku as the graphenedb environment variables changed names."

This reverts commit 1d203d95ecf60d47c824b9a9518ad8fa6b8b7583.

beercanx committed:
[CI] Removal of obsolete matrix testing of neo4j version.

beercanx committed:
[Clean Up] Removal of some vagrant configuration as its no longer used.

beercanx committed:
[Docker] Updating the compose configuration to start up the same as Heroku.

beercanx committed:
[Dependency] Updating small affecting dependencies.

beercanx committed:
[Dependencies] Updating spring boot and neo4j.

beercanx committed:
[Dependencies] Updating maven plugin versions.

beercanx committed:
[CleanUp] Removal of unnessesary code.

beercanx committed:
[IntegrationTests] Adding tests to cover fining a password for a given user.

beercanx committed:
[API] Adding a fix to cause a not found exception when we're unable to find a password for a given user.

beercanx committed:
[UnitTest] Adding paginated results test to verify full page abilities.

beercanx committed:
[UnitTest] Adding test to verify disabled and failing SchemaConfiguration performs.

beercanx committed:
[UnitTest] Adding some unit tests around error handling in the symmetric engine.

beercanx committed:
[Ciphers] Minor code improvements, only init a secure random once.

beercanx committed:
[Ciphers] Exposed private methods for tests. Removed unused code path.

beercanx committed:
[UnitTests] Added some tests around the SymmetricGenerator.

beercanx committed:
[UnitTests] Updated a class to provide a test constructor.

beercanx committed:
[CodeQuality] Removal of unused import in test.

beercanx committed:
[Utilities] Adding throwing functions into main code base.

beercanx committed:
[UnitTests] Making the symmetic generator more testable with a key gen factory.

beercanx committed:
[UnitTest] Updating unit tests to take advantage of key gen factory interface.

beercanx committed:
[CodeFiddling] More specific exceptions

in the SymmetricEngine

beercanx committed:
[BugFix] Correctly generating the different param specs.

beercanx committed:
[UnitTests] GCM coverage SymmeticEngine.

, both encrypt and decrypt.

beercanx committed:
[UnitTest] Updating test with different data.

beercanx committed:
[UnitTests] Thowing function unit tests.

beercanx committed:
Update to Lombok.

beercanx committed:
Updating dependency versions.

beercanx committed:
Additional cofig for local running.

beercanx committed:
Reverted the Neo4J upgrade, as it breaks things within the Spring stack.

beercanx committed:
[Lint] Removal of unused imports to improve linting.

beercanx committed:
Updating unit test to be more precise.

beercanx committed:
Created an embedded version of the API. Also now create a bundled versions to enable a compact embedded version.

beercanx committed:
A start on a password service to first handle secure creations.

beercanx committed:
Patching BouncyCastle.

beercanx committed:
Addition of a PasswordService that can create and share a secured password with basic validation. Also started to extract URL building.

beercanx committed:
Fix to tests around Ciphers and PowerMocking.

beercanx committed:
Lint fix

beercanx committed:
[Test] Updating IT with recent changes around create password.

beercanx committed:
[Test] Updating IT with static reference to base password endpoint.

beercanx committed:
[Service] Adding validaton messages to the password creation.

beercanx committed:
[Model] Updating the password model to be equal only when UUIDs match.

beercanx committed:
Updating to latest software where possible.

beercanx committed:
auto lint formatting.

beercanx committed:
Marking lombok as a provided dependency, due to it being only required at compile time.

beercanx committed:
More lint and reverting a bug.

beercanx committed:
Initial start at actually encrypting the password on creation and save.

beercanx committed:
Auto lint formatting.

Commit:
Added another version to test but restricted to bolt.

Commit:
Adding more versions of Neo4j to test and removal of commented out config.

Commit:
Updated embedded version to be same as Heroku instance.

Commit:
Updating NewRelic version for Heroku, also updating spring boot version.

Commit:
Updating to latest failsafe as the problem is supposed to be fixed.

beercanx committed:
New locations and uri builder

beercanx committed:
Using common locations helper

beercanx committed:
Basic style formatting auto applied

beercanx committed:
Corrected tests

beercanx committed:
Stopped hiding errors in tests with runtimes

beercanx committed:
Stopped using runtime exception in builder

beercanx committed:
Change to importing of statics

beercanx committed:
Remove unused variable.

beercanx committed:
removed unused import

beercanx committed:
Updated travis to attempt on Java 9

beercanx committed:
Revert "Updated travis to attempt on Java 9"

This reverts commit 25d7033dbf66e9b245e8b51bd7246a08ac7b25fe.

beercanx committed:
Adding slack integration.
